### ISSUE TITLE
88.9%-Accuracy about recognition—MS1MV3-Resnet50

### ISSUE THREAD

**Issue by I1amok** [deepinsight/insightface#1892]
I did a test with the LFW dataset，however pretrained-MS1MV3-Resnet50 which the accuracy is only 88.9.
The same test showed different results on other data, with an accuracy rate of at least 99.8.
I don't know why, hope someone will test it to see if it has the same problem

**Comment by I1amok** [comment]
I randomly selected 3000 LFW data, in which the ratio of identical ids to different ids was 1:1. The pre-processing weights of MS1MMv3-RESnet50 were downloaded from the official source and loaded with the weights of the model through inference. Py in the official 2021 PyTorch code. Through the test code written by myself, the accuracy rate is only 0.89. To verify if there was a problem with my test code, I went to test the weights trained by others in other ArcFace code, and the results were completely different, their accuracy was reasonable at 0.98+.

**Comment by I1amok** [comment]
<img width="880" alt="EA7216F5-D695-4F14-92D2-51A91955AAA5" src="https://user-images.githubusercontent.com/95618895/153533285-e0309120-9d26-493a-9123-bab44be5a07f.png"><img width="889" alt="3D53D51D-12A9-4D27-B4F9-0F83D0BE89CE" src="https://user-images.githubusercontent.com/95618895/153533314-8ab1aeb1-6575-4fa7-ae53-e9f1f6333fe3.png">

**Comment by I1amok** [comment]
@NaimaBousnina ,I am very sorry that I forgot to reply you in time because I was busy recently.I found two copies of THE LFW test set yesterday, and their sizes are 128*128, 96*112 respectively. I think this may be the main reason for the difference in accuracy, and I will further confirm this later.I will keep you informed of any new results.As for your problem, I wonder if you have solved it. If not, you can contact me by email, and I will send you a relevant document for reference.My email account number is 772769720@qq.com.

**Comment by imxtx** [comment]
@I1amok Have you solved this problem? I also got similar test accuracy on LFW which is 90.12. I prepared the LFW dataset with MTCNN. Here is how I preprocessed the lfw-deepfunneled dataset:

```python
from facenet_pytorch import MTCNN
from pathlib import Path
from PIL import Image
from tqdm import tqdm
import os


if __name__ == '__main__':

    # use MTCNN to detect, crop, and save
    margin = 60
    mtcnn = MTCNN(margin=margin, post_process=False,
                  device='cuda', thresholds=[0.5, 0.6, 0.7])

    # set up paths
    base = '/home1/txxie/datasets/LFW/'
    deepfunneled = 'lfw-deepfunneled'
    aligned160 = f'deepfunneled-align160-margin{margin}'
    file_list = 'file_list_all.txt'
    Path(os.path.join(base, aligned160)).mkdir(parents=True, exist_ok=True)

    with open(os.path.join(base, file_list)) as f:
        for line in tqdm(f.readlines()):
            img_path = line.strip()
            abs_img_path = os.path.join(base, deepfunneled, img_path)
            save_path = os.path.join(base, aligned160, img_path)
            # print(f'processing {save_path}')
            image = Image.open(abs_img_path)
            mtcnn(image, save_path=save_path)

    # for margin in tqdm([0, 10, 20, 30, 40, 50]):
    #     mtcnn = MTCNN(margin=margin, post_process=False,
    #                   device='cuda', thresholds=[0.5, 0.6, 0.7])
    #     aligned160 = f'deepfunneled-align160-margin{margin}'
    #     # cannot detect this image with default thresholds
    #     img_path = 'Marilyn_Monroe/Marilyn_Monroe_0001.jpg'
    #     abs_img_path = os.path.join(base, deepfunneled, img_path)
    #     save_path = os.path.join(base, aligned160, img_path)
    #     # print(f'processing {save_path}')
    #     image = Image.open(abs_img_path)
    #     mtcnn(image, save_path=save_path)
```
